### PR TITLE
[EA] People directory sorting dropdown

### PR DESCRIPTION
**Pull request by oetherington**
Adds a sorting dropdown and a new "default" sorting option to the people directory. [Relevant Figma](https://www.figma.com/design/qAHRGADVOGojqKpqbGMFFu/2024-Q2?node-id=685-6299&m=dev).

The default option sorts by relevance when a search query is entered or by a new "profile completion" field when there is no query.

Profile completion is calculated as the sum of the points for the following completed fields:
 - 1 point for each of profile image, job title, organization, biography, how others can help me, how I can help others, career stage and map location
 - 0.25 points for each of website, Github URL, Twitter URL, Facebook URL and LinkedIn URL
 - Plus one point if you've ever written a comment
 - Plus two points if you've ever written a post
 - A linear weighting factor is then added based on how much karma you have

Screenshot 2024-06-25 at 12 57 07

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1207649400448921) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by oetherington** [comment]
@agnesstenlund

**Comment by agnesstenlund** [comment]
Looks great!! Weighting sounds good too, thanks for putting so much care into that. It feels so much better viewing the page with the default sort 😍 makes it feel like the Forum is an active place

- [ ] I noticed there are no hover states in the dropdown (this is likely cause I forgot about adding that to the spec). I was imagining something like this (we use this for other dropdowns elsewhere)
Screenshot 2024-06-26 at 00 21 39

- [ ] Would also probably say "Sort" instead of "Sorting", unless you changed it for a reason (like "sorting" being more common or more correct english)
Screenshot 2024-06-26 at 00 23 57

File: packages/lesswrong/components/peopleDirectory/PeopleDirectoryFilters.tsx [hunk 1]
```diff
@@ -69,21 +75,49 @@ export const PeopleDirectoryFilters = ({classes}: {
...
+              icon="ArrowsUpDown"
+              style="button"
+              className={classes.columnsList}
+            >
+              <PeopleDirectoryCheckOption
+                label="Default"
+                selected={isDefaultSorting}
+                onSelect={setSorting.bind(null, null)}
+              />
+              {columns.map(({label, sortField, defaultSort}) => sortField
```
Comment (Will-Howard) [inline packages/lesswrong/components/peopleDirectory/PeopleDirectoryFilters.tsx:90]: It's a bit confusing that there is no column for "Location" or "Profile last updated" when you sort by them. I think if it added the column of the thing you are sorting by that would be a lot better (although still a bit confusing)

The comments sorting is also a quite confusing (and also didn't work for me due to an elastic error "No mapping found for [commentCount] in order to sort on"). I would favour either showing the number of comments or just not having that sorting if there's a reason we don't want to do that

File: packages/lesswrong/components/peopleDirectory/PeopleDirectoryFilters.tsx [hunk 2]
```diff
@@ -69,21 +75,49 @@ export const PeopleDirectoryFilters = ({classes}: {
...
+                ? (
+                  <PeopleDirectoryCheckOption
+                    key={label}
+                    label={label}
+                    selected={sortField === sorting.field}
+                    onSelect={setSorting.bind(null, {
+                      field: sortField,
+                      direction: defaultSort ?? "asc",
+                    })}
```
Comment (Will-Howard) [inline packages/lesswrong/components/peopleDirectory/PeopleDirectoryFilters.tsx:99]: I would like clicking an already checked option to switch back to the default sorting

**Review by Will-Howard — APPROVED** [review]
With no query or sorting I'm getting a "No mapping found for [profileCompletion] in order to sort on" error which is causing the page not to load:
Screenshot 2024-07-12 at 12 52 20

Presumably this is due to needing to re-sync the elastic documents. It did work with sortings applied so I'm calling this ASTLAC, but could you make sure that isn't a real issue.

None of my other comments are blocking but I think the one about making sure the column you are sorting by exists in the table would be good to change

**Comment by oetherington** [comment]
@Will-Howard Just finishing up the last changes now, then we should be able to merge all the remaining people directory stuff.

RE the "No mapping found" problem - are you sure you were using the dev DB? I can't repro on dev, but I'd expect this error with the staging or prod DB.

**Comment by Will-Howard** [comment]
> RE the "No mapping found" problem - are you sure you were using the dev DB? I can't repro on dev, but I'd expect this error with the staging or prod DB.

@oetherington I thought I tested it on both dev and prod, but anyway it's working now on dev so either I didn't or some re-syncing has fixed it

**Pull request merged**